src/KafkaFlow.Microsoft.DependencyInjection/MicrosoftDependencyResolver.cs

```diff
@@ -15,7 +15,7 @@ public MicrosoftDependencyResolver(IServiceProvider serviceProvider)
 
         public object Resolve(Type type)
         {
-            return this.serviceProvider.GetRequiredService(type);
+            return this.serviceProvider.GetService(type);
```

Comment:
why this change?

Reply:
When a type that was not registered is requested, `GetRequiredService` throws an exception, whilst `GetService` returns null.
`AddApacheAvroSerializer` extensions need to try to resolve `ISchemaRegistryClient` and send the instance (or null) to `ApacheAvroMessageSerializer`. 
If there is no `ISchemaRegistryClient` instance, then `ApacheAvroMessageSerializer` will thrown a proper exception.

Reply:
Another approach would be to keep `GetRequiredService` and use a try/catch block when trying to get the `ISchemaRegistryClient` instance in the `AddApacheAvroSerializer` extension. 
IMHO, it would be the best approach, what do you think?